packages/ui/src/components/TransferButton.tsx

```diff
@@ -14,33 +14,38 @@ interface Props {
 }
 
 export function TransferButton({ from, to, disabled }: Props) {
-  const [isOpen, toggleOpen] = useToggle()
+  const { showModal } = useModal()
   const isTransfer = !from && !to
   const isSend = !!from && !isTransfer
-  const icon = isTransfer ? <TransferIcon /> : isSend ? <ArrowOutsideIcon /> : <ArrowInsideIcon />
+  const iconName: TransferIconName = isTransfer ? 'TransferIcon' : isSend ? 'SendIcon' : 'ReceiveIcon'
```

Comment:
Let's extract this logic to have:

```
type TransferType: 'Send' | 'Receive' | 'Transfer'
const getType( from: boolean, to: boolan ) => TransferType
```

Having this we don't have to push `icon` to the modal and we can have the same component that renders different icons depending on the type.

```
<TransferIcon type={getIconType(true, false)} />
```

Another thing is to update the button accordingly.